WatchParty/client/src/components/Header/Header.js

```diff
@@ -1,43 +1,96 @@
 import React from 'react';
+import { Route, Switch, withRouter} from "react-router-dom"
 import { Nav, Navbar, NavDropdown } from 'react-bootstrap'
 import { Form, FormControl, Button } from 'react-bootstrap'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 
+import SearchResults from "./SearchResults"
+
 import './Header.css'
-import { useAuth0 } from "../../react-auth0-spa";
-
-
-const Header = () => {
-  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();
-  return (
-    <Navbar collapseOnSelect expand="md" bg="dark" variant="dark">
-      <Navbar.Brand href="#home">WatchParty</Navbar.Brand>
-      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
-      <Navbar.Collapse id="responsive-navbar-nav">
-        <Nav className="mr-auto">
-          <Form inline>
-            <FormControl type="text" placeholder="Search movies or users" className="mr-sm-2" />
-            <Button variant="outline-info">Search</Button>
-          </Form>
-        </Nav>
-        <Nav>
-          {!isAuthenticated && (
-            <Nav.Link onClick={() => loginWithRedirect({})}>Log in / Sign up</Nav.Link>
-          )}
-          {isAuthenticated && (
-            <NavDropdown
-              title=<FontAwesomeIcon icon='user-circle' size='2x' />
-              id="collapsible-nav-dropdown"
-              alignRight>
-              <NavDropdown.Item href="#profile">Profile</NavDropdown.Item>
-              <NavDropdown.Divider />
-              <NavDropdown.Item onClick={() => logout()}>Log out</NavDropdown.Item>
-            </NavDropdown>
-          )}
-        </Nav>
-      </Navbar.Collapse>
-    </Navbar>
-  )
+
+class Header extends React.Component {
+ 
+  constructor(props){
+    super(props);
+
+    this.state ={
+      searchText: ''
+    };
+
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.handleTextChange = this.handleTextChange.bind(this);
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+    this.handleSearchKeyUp = this.handleSearchKeyUp.bind(this);
+  }
+
+  handleRoute = route => () => {
+    this.props.history.push({ pathname: route });
+  };
+
+  handleTextChange(event) {
+    this.setState( {searchText: event.target.value} );
+  }
+
+  handleSearchSubmit(event){
+    if(this.state.searchText){
+      let temp = this.state.searchText;
+      this.setState( {searchText: ''} )
+
+      this.props.history.push({
+        pathname: "/results",
+        state: {searchText: temp}
+      });
+
+    } else {
+      alert("Please enter a movie name!")
+    }
+
+  }
+
+  handleFormSubmit(event){
+    event.preventDefault();
+  }
```

Comment:
Couldn't this be implemented as a one-liner, in the render() method?

Reply:
fixed